stdlib/_interpreters.pyi

```diff
@@ -0,0 +1,48 @@
+import types
+from collections.abc import Callable, Mapping
+from typing import Final, Literal, SupportsIndex, final
+from typing_extensions import TypeAlias
+
+_Configs: TypeAlias = Literal["default", "isolated", "legacy", "empty"]
+
+class InterpreterError(Exception): ...
+class InterpreterNotFoundError(InterpreterError): ...
+class NotShareableError(Exception): ...
+
+# Not-instantiable, so mark as final.
+@final
+class CrossInterpreterBufferView:
+    def __buffer__(self, flags: int, /) -> memoryview: ...
+
+def new_config(name: _Configs = "isolated", /, **overides: object) -> types.SimpleNamespace: ...
+def create(config: types.SimpleNamespace | _Configs | None = "isolated", *, reqrefs: bool = False) -> int: ...
+def destroy(id: SupportsIndex, *, restrict: bool = False) -> None: ...
+def list_all(*, require_ready: bool) -> list[tuple[int, int]]: ...
+def get_current() -> tuple[int, int]: ...
+def get_main() -> tuple[int, int]: ...
+def is_running(id: SupportsIndex, *, restrict: bool = False) -> bool: ...
+def get_config(id: SupportsIndex, *, restrict: bool = False) -> types.SimpleNamespace: ...
+def whence(id: SupportsIndex) -> int: ...
+def exec(id: SupportsIndex, code: str, shared: bool | None = None, *, restrict: bool = False) -> None: ...
+def call(
+    id: SupportsIndex,
+    callable: Callable[..., object],
+    args: tuple[object, ...] | None = None,
+    kwargs: dict[str, object] | None = None,
+    *,
+    restrict: bool = False,
+) -> object: ...
+def run_string(id: SupportsIndex, script: str, shared: bool | None = None, *, restrict: bool = False) -> None: ...
+def run_func(id: SupportsIndex, func: Callable[..., object], shared: bool | None = None, *, restrict: bool = False) -> None: ...
```

Comment:
`func` can be either a zero-argument function (`Callable[[], object]`) or a code object.

Reply:
https://github.com/python/typeshed/pull/12230/commits/9cd08daaf26beb6c2ad3cac3cc888078a1957fb4